src/v/raft/consensus.cc

```diff
@@ -1383,10 +1381,32 @@ ss::future<> consensus::do_start() {
           "Configuration manager started: {}",
           _configuration_manager);
 
-        co_await _offset_translator.start(
-          offset_translator::must_reset{initial_state});
-
-        co_await _snapshot_lock.with([this] { return hydrate_snapshot(); });
```

Comment:
I'm not sure that holding `_snapshot_lock` is important here (`write_snapshot` probably can't be called at this time) but wouldn't hurt I guess?

Reply:
Sure, done